### PR TITLE
Unhardcode the palette the missile weapon of the NukePower uses

### PR DESCRIPTION
**Pull request by abcdefg30**
Just some unhardcoding and documenting.

### CODE REVIEW

**Comment by pchote** [comment]
I'm not convinced its worth the effort to maintain/improve/document this code considering our long-standing plan to throw it all out (ref comment in #7349 and several discussions in IRC).

If you need to customize the palette for RA2, could you consider taking on the task of fixing this properly? @GraionDilach was going to take this on, but it has been several months since I've heard anything from him on this.

**Comment by abcdefg30** [comment]
I already have a certain branch (which I showed you some time ago) waiting to be finished: https://github.com/OpenRA/OpenRA/compare/bleed...abcdefg30:supportoverlay (see http://logs.openra.net/?year=2015&month=10&day=17#18:06:42).
But until that is finished I think this doesn't hurt.

**Comment by abcdefg30** [comment]
From what I remember, @GraionDilach abandoned the work on this for now and would be happy if I take over, but don't quote me on that, I also don't find the IRC log at the moment.

File: OpenRA.Mods.Common/Traits/SupportPowers/NukePower.cs [hunk 1]
```diff
@@ -21,9 +21,17 @@ namespace OpenRA.Mods.Common.Traits
...
+		[Desc("Weapon to use for the inpact.",
+			"But also image to use for the missile.",
+			"Requires an 'up' and a 'down' sequence on the image.")]
 		public readonly string MissileWeapon = "";
+
+		[Desc("Offset from the actor the missile spawns on.")]
 		public readonly WVec SpawnOffset = WVec.Zero;
 
+		[Desc("Palette to use for the missle weapon image.")]
+		public readonly string MissilePalette = "effect";
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/SupportPowers/NukePower.cs]: `[PaletteReference]` please.

**Comment by Mailaender** [comment]
:+1: Note that this doesn't mean that I in any way object the grand plan proposed over a year ago.

**Comment by GraionDilach** [comment]
I literally got mad at missile beacon calculation and had no time since to come up with something for that (#9827 is where I've got and the next step was to create a projectile type for nukes).

**Comment by abcdefg30** [comment]
Added the `[PaletteReference]`.

File: OpenRA.Mods.Common/Traits/SupportPowers/NukePower.cs [hunk 2]
```diff
@@ -21,9 +21,17 @@ namespace OpenRA.Mods.Common.Traits
 	class NukePowerInfo : SupportPowerInfo, IRulesetLoaded, Requires<BodyOrientationInfo>
 	{
 		[WeaponReference]
+		[Desc("Weapon to use for the inpact.",
```
Comment (Phrohdoh) [inline OpenRA.Mods.Common/Traits/SupportPowers/NukePower.cs]: `impact`

Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/SupportPowers/NukePower.cs]: Ops, fixed.

File: OpenRA.Mods.Common/Traits/SupportPowers/NukePower.cs [hunk 3]
```diff
@@ -21,9 +21,17 @@ namespace OpenRA.Mods.Common.Traits
...
 		[WeaponReference]
+		[Desc("Weapon to use for the impact.",
+			"But also image to use for the missile.",
+			"Requires an 'up' and a 'down' sequence on the image.")]
 		public readonly string MissileWeapon = "";
+
+		[Desc("Offset from the actor the missile spawns on.")]
 		public readonly WVec SpawnOffset = WVec.Zero;
 
+		[Desc("Palette to use for the missle weapon image.")]
```
Comment (obrakmann) [inline OpenRA.Mods.Common/Traits/SupportPowers/NukePower.cs]: "missle"

**Comment by obrakmann** [comment]
The wording is a bit awkward in the descriptions, but it'll do. :+1:

**Comment by GraionDilach** [comment]
IIRC Westwood also typoed it Missle on the cameo.

File: OpenRA.Mods.Common/Traits/SupportPowers/NukePower.cs (hunk 3 above)
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/SupportPowers/NukePower.cs]: Derp.

**Comment by abcdefg30** [comment]
Fixed.

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/420e400338df4fc1c880bcf7a003958562da127f...b7af480b708c5c2b624a1842f572b0cda41e9625)

**Pull request merged**